core/src/main/java/com/scalar/db/storage/objectstorage/SelectStatementHandler.java

```diff
@@ -0,0 +1,267 @@
+    operation.checkArgument(Get.class);
+    ObjectStoragePartition partition =
+        getPartition(getNamespace(get), getTable(get), operation.getConcatenatedPartitionKey());
+    if (!partition.getRecord(operation.getRecordId()).isPresent()) {
+      return new EmptyScanner();
+    }
+    ObjectStorageRecord record = partition.getRecord(operation.getRecordId()).get();
+    return new ScannerImpl(
+        Collections.singletonList(record).iterator(),
+        new ResultInterpreter(get.getProjections(), metadata),
+        1);
+  }
+
+  private Scanner executeScan(Scan scan, TableMetadata metadata) throws ExecutionException {
+    ObjectStorageOperation operation = new ObjectStorageOperation(scan, metadata);
+    operation.checkArgument(Scan.class);
+    ObjectStoragePartition partition =
+        getPartition(getNamespace(scan), getTable(scan), operation.getConcatenatedPartitionKey());
+    List<ObjectStorageRecord> records = new ArrayList<>(partition.getRecords().values());
+
+    records.sort(
+        (o1, o2) ->
+            new ClusteringKeyComparator(metadata)
+                .compare(o1.getClusteringKey(), o2.getClusteringKey()));
+    if (isReverseOrder(scan, metadata)) {
+      Collections.reverse(records);
+    }
+
+    // If the scan is for DESC clustering order, use the end clustering key as a start key and the
+    // start clustering key as an end key
+    boolean scanForDescClusteringOrder = isScanForDescClusteringOrder(scan, metadata);
+    Optional<Key> startKey =
+        scanForDescClusteringOrder ? scan.getEndClusteringKey() : scan.getStartClusteringKey();
+    boolean startInclusive =
+        scanForDescClusteringOrder ? scan.getEndInclusive() : scan.getStartInclusive();
+    Optional<Key> endKey =
+        scanForDescClusteringOrder ? scan.getStartClusteringKey() : scan.getEndClusteringKey();
+    boolean endInclusive =
+        scanForDescClusteringOrder ? scan.getStartInclusive() : scan.getEndInclusive();
+
+    if (startKey.isPresent()) {
+      records =
+          filterRecordsByClusteringKeyBoundary(
+              records, startKey.get(), true, startInclusive, metadata);
+    }
+    if (endKey.isPresent()) {
+      records =
+          filterRecordsByClusteringKeyBoundary(
+              records, endKey.get(), false, endInclusive, metadata);
+    }
+
+    if (scan.getLimit() > 0) {
+      records = records.subList(0, Math.min(scan.getLimit(), records.size()));
+    }
+
+    return new ScannerImpl(
+        records.iterator(),
+        new ResultInterpreter(scan.getProjections(), metadata),
+        scan.getLimit());
+  }
+
+  private Scanner executeScanAll(ScanAll scan, TableMetadata metadata) throws ExecutionException {
+    ObjectStorageOperation operation = new ObjectStorageOperation(scan, metadata);
+    operation.checkArgument(ScanAll.class);
+    Set<ObjectStorageRecord> records = getRecordsInTable(getNamespace(scan), getTable(scan));
```

Comment:
It looks like this logic loads all records of the table into memory, right? If so, can we change it to process the records in a streaming manner instead?